### PR TITLE
Feat/Small frontend features

### PR DESCRIPTION
**Pull request by SimonSta95**
Added Breadcrumbs, Textsearch and Filtering by platform

### CODE REVIEW

File: gamehub-frontend/src/pages/GameGallery/GameGallery.tsx [hunk 1]
```diff
@@ -1,16 +1,25 @@
...
+};
 
 export default function GameGallery(props: Readonly<GameGalleryProps>) {
+    const [searchQuery, setSearchQuery] = useState('');
+    const [selectedPlatform, setSelectedPlatform] = useState<string>('');
+
+    const platforms = Array.from(new Set(props.games.flatMap(game => game.platforms)));
+
+    const filteredGames = props.games.filter(game =>
+        game.title.toLowerCase().includes(searchQuery.toLowerCase()) &&
```
Comment (rinaehyun) [inline gamehub-frontend/src/pages/GameGallery/GameGallery.tsx:20]: Just one small tipp:
If there is any chance that 'game.title' or 'searchQuery' could be null or undefined, please use '?'.
`game.title?.toLowerCase().includes(searchQuery?.toLowerCase() || '') &&`

If it is certain that  'game.title' or 'searchQuery' will always be non-null strings, there is no need to use '?'.

**Review by rinaehyun — APPROVED** [review]
Looks good to me!

File: gamehub-frontend/src/pages/GameGallery/GameGallery.tsx (hunk 1 above)
Comment (rinaehyun) [inline gamehub-frontend/src/pages/GameGallery/GameGallery.tsx]: a small tipp:
If there's any chance that game.title or searchQuery could be null or undefined, please use '?'
`game.title?.toLowerCase().includes(searchQuery?.toLowerCase() || '') && `

If it is certain that 'game.title' or 'searchQuery' has always non-null strings, don't need to use '?'

Comment (SimonSta95) [reply gamehub-frontend/src/pages/GameGallery/GameGallery.tsx]: thanks  :)

Comment (rinaehyun) [reply gamehub-frontend/src/pages/GameGallery/GameGallery.tsx]: You're welcome :)

**Pull request merged**